Lib/test/test_urllib2.py

```diff
@@ -802,6 +802,41 @@ def connect_ftp(self, user, passwd, host, port, dirs,
         else:
             self.fail("Did not raise ftplib exception")
 
+    def test_ftp_error_closes_connection(self):
+        #Test that FTP connection is closed when retrfile() raises an error.
+        #See https://github.com/python/cpython/issues/140691
+        import warnings
+        
+        mock_fw = mock.MagicMock()
+        mock_fw.retrfile.side_effect = ftplib.error_temp("Connection timed out")
+
+        with mock.patch.object(urllib.request.FTPHandler, 'connect_ftp', 
+                               return_value=mock_fw):
+            # Create a mock request with required attributes
+            req = mock.MagicMock()
+            req.get_full_url.return_value = 'ftp://example.com/test.txt'
+            req.host = 'example.com'
+            req.selector = 'test.txt'
+            req.type = 'ftp'
+            req.timeout = 5
+
+            handler = urllib.request.FTPHandler()
+
+            with warnings.catch_warnings(record=True) as warning_list:
```

Comment:
This is part of the testing needed, will also need to use `test.support` to check that there are no "unraisable exceptions" (which is what this one ends up being when it isn't correctly cleaned up). Look at other tests for examples